contracts/BackedAutoFeeTokenImplementation.sol

```diff
@@ -186,13 +213,15 @@ contract BackedAutoFeeTokenImplementation is BackedTokenImplementation {
         public
         view
         virtual
-        returns (uint256 newMultiplier, uint256 periodsPassed)
+        returns (uint256 newMultiplier, uint256 periodsPassed, uint256 newMultiplierNonce)
     {
         periodsPassed = (block.timestamp - lastTimeFeeApplied) / periodLength;
         newMultiplier = multiplier;
+        newMultiplierNonce = multiplierNonce;
         if (feePerPeriod > 0) {
             for (uint256 index = 0; index < periodsPassed; index++) {
                 newMultiplier = (newMultiplier * (1e18 - feePerPeriod)) / 1e18;
+                newMultiplierNonce += 1;

```

Comment:
I believe it would be just more efficient to move it out of the loop and increase that by periodsPassed.